egs/wsj/s5/steps/conf/train_calibration.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+# Copyright 2015, Brno University of Technology (Author: Karel Vesely). Apache 2.0.
+
+# Trains logistic regression, which calibrates the per-word confidences in 'CTM'.
+# The 'raw' confidences are obtained by Minimum Bayes Risk decoding.
+
+# The input features of logistic regression are:
+# - logit of Minumum Bayer Risk posterior
+# - log of word-length in characters
+# - log of average-depth depth of a lattice at words' position
+# - log of frames per character ratio
+# (- categorical distribution of 'lang/words.txt', DISABLED)
+
+# begin configuration section.
+cmd=
+lmwt=12
+decode_mbr=true
+word_min_count=10 # Minimum word-count for single-word category,
+normalizer=0.0025 # L2 regularization constant,
+stage=0
+# end configuration section.
+
+[ -f ./path.sh ] && . ./path.sh
+. parse_options.sh || exit 1;
+
+if [ $# -ne 7 ]; then
+  echo "Usage: $0 [opts] <data-dir> <lang-dir|graph-dir> <word-filter> <word-length> <unigrams> <decode-dir> <calibration-dir>"
```

Comment:
Well, IMHO it is more `readable' to have one feature per file, while the file-name `says' what it is. For example the filtering flag is not even an input feature for logistic regression. Then, some features don't depend only on the word itself, for example : 'frame_length / word_length' needs the word duration, which can't be accommodated by generic look-up table for 'some word-specific features'. On the other hand it increases the complexity of the 'loclal/' script, which needs to prepare more complicated input. I see that possibility to freely add some word-specific features is interesting, but more interesting would be to be able to add some 'dynamic' features no the word-identities. However this implies a modification in the 'steps/conf/prepare_calibration_data.py', which is the only place where information from various sources can be freely combined. So if we adopt the format:
integer-word-id feature1 feature2 ... featureN
we need to implicitly hard-wire the meaning of the columns, which can cause confusion if somebody later on modifies the code. Do you still want to adopt it? I think it is not a good idea, but still I can change it that way...